src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -686,38 +691,54 @@ public async Task<VersionStamp> GetSemanticVersionAsync(CancellationToken cancel
                 analyzerConfigSet ?? _lazyAnalyzerConfigOptions);
         }
 
-        private ProjectInfo.ProjectAttributes Attributes
+        internal ProjectInfo.ProjectAttributes Attributes
             => ProjectInfo.Attributes;
 
-        private ProjectState WithAttributes(ProjectInfo.ProjectAttributes attributes)
+        private ProjectState WithRawAttributes(ProjectInfo.ProjectAttributes attributes)
```

Comment:
Seems like this needs a comment compared to WithAttributes explaining why there's two now. If the assumption is the attributes here passed in already had their version changed, then that could be asserted (or else we might have a bug?). But then why can't we just move all the callers to the other WithAttributes?

Or should this be WithAttributesAlreadyHavingNewerVersion or something?